packages/react-native-web/src/exports/Text/types.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) Nicolas Gallagher.
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type { ColorValue, GenericStyleProp, LayoutEvent } from '../../types';
+import type { ViewStyle } from '../View/types';
+
+type FontWeightValue =
+  | 'normal'
+  | 'bold'
+  | '100'
+  | '200'
+  | '300'
+  | '400'
+  | '500'
+  | '600'
+  | '700'
+  | '800'
+  | '900';
+
+export type TextStyle = {
+  ...ViewStyle,
+  color?: ColorValue,
+  fontFamily?: string,
+  fontFeatureSettings?: string,
+  fontSize?: number | string,
+  fontStyle?: 'italic' | 'normal',
+  fontWeight?: FontWeightValue,
+  fontVariant?: $ReadOnlyArray<
+    'small-caps' | 'oldstyle-nums' | 'lining-nums' | 'tabular-nums' | 'proportional-nums'
+  >,
+  letterSpacing?: number | string,
+  lineHeight?: number | string,
+  textAlign?: 'center' | 'end' | 'inherit' | 'justify' | 'justify-all' | 'left' | 'right' | 'start',
+  textAlignVertical?: string,
+  textDecorationColor?: ColorValue,
+  textDecorationLine?: 'none' | 'underline' | 'line-through' | 'underline line-through',
+  textDecorationStyle?: 'solid' | 'double' | 'dotted' | 'dashed',
+  textIndent?: number | string,
+  textOverflow?: string,
+  textRendering?: 'auto' | 'geometricPrecision' | 'optimizeLegibility' | 'optimizeSpeed',
+  textShadowColor?: ColorValue,
+  textShadowOffset?: {| width?: number, height?: number |},
+  textShadowRadius?: number,
+  textTransform?: 'capitalize' | 'lowercase' | 'none' | 'uppercase',
+  unicodeBidi?: 'normal' | 'bidi-override' | 'embed' | 'isolate' | 'isolate-override' | 'plaintext',
+  whiteSpace?: string,
+  wordBreak?: 'normal' | 'break-all' | 'break-word' | 'keep-all',
+  wordWrap?: string,
+  writingDirection?: 'auto' | 'ltr' | 'rtl',
+  /* @platform web */
+  MozOsxFontSmoothing?: string,
+  WebkitFontSmoothing?: string
```

Comment:
Follow up: consider using the stylesheet compiler to prefix `fontSmoothing` instead of exposing these properties